### PR TITLE
store download directory and store list path

### PR DESCRIPTION
**Pull request by markwinter**
This makes two changes:

- `store list` now takes an optional path to list a certain directory in the store
- `store download <path>` now can take a directory path and it will download everything in the directory

```
$ ~/Code/cli/unweave store download data/MNIST
t10k-images-idx3-ubyte 0.00 b / 7.48 MiB [----------------------------------------------------------] 0s ] 0.00 b/s
t10k-labels-idx1-ubyte saved to ./uwstore/data/MNIST/raw/t10k-labels-idx1-ubyte
train-images-idx3-ubyte 15.06 MiB / 44.86 MiB [==================>---------------------------------------] 1s ] 25.15 MiB/s
train-labels-idx1-ubyte saved to ./uwstore/data/MNIST/raw/train-labels-idx1-ubyte

```

```
$ ~/Code/cli/unweave store list data/MNIST/
2022-08-26 22:05:38 +0900 KST	7.8 MB	data/MNIST/raw/t10k-images-idx3-ubyte
2022-08-26 22:05:39 +0900 KST	10 kB	data/MNIST/raw/t10k-labels-idx1-ubyte
2022-08-26 11:15:03 +0900 KST	47 MB	data/MNIST/raw/train-images-idx3-ubyte
2022-08-26 11:15:08 +0900 KST	60 kB	data/MNIST/raw/train-labels-idx1-ubyte
```

Resolves UNW-474 UNW-485

### CODE REVIEW

File: controller/store.go [hunk 1]
```diff
@@ -29,12 +29,22 @@ func (c *Controller) StoreList(ctx context.Context) error {
...
-	objects, err := c.api.ListProjectStore(ctx, projectID)
+	objects, err := c.api.ListProjectStore(ctx, projectID, path)
 	if err != nil {
 		return fmt.Errorf("failed to list project store: %v", err)
 	}
 
 	for _, object := range objects {
+		// If we listed an actual object above, not a dir, it returns a single item
+		// with an empty path, so set the object path to the path that the user gave
+		if object.Path == "" {
```
Comment (markwinter) [inline controller/store.go:40]: Thinking about adding this to the workbench instead

Comment (markwinter) [reply controller/store.go:40]: Will require a breaking change to how listing in dashboard so will be done later.

**Comment by noorvir** [comment]
Is this implemented on the `api` and `workbench` side?

**Comment by markwinter** [comment]
Yes that's what was just merged https://github.com/unweave/unweave/pull/459

**Review by noorvir — CHANGES_REQUESTED** [review]

File: controller/store.go [hunk 2]
```diff
@@ -54,39 +68,102 @@ func (c *Controller) StoreDownload(ctx context.Context, objectPath string) error
...
+			defer wg.Done()
+			if err := c.downloadObject(ctx, projectID, obj, bar); err != nil {
+				errs = append(errs, err)
+			}
+		}(object)
+	}
+
+	printer.Wait()
+
+	for _, err := range errs {
```
Comment (markwinter) [inline controller/store.go:120]: We print after because any printing during would mess up the other download bars

**Comment by markwinter** [comment]
Was very simple to add an `store ls` alias for `store list` with cobra. Added based on that user's feedback.

File: main.go [hunk 3]
```diff
@@ -120,10 +120,11 @@ func init() {
 		Args:  cobra.NoArgs,
 	}
 	storeCmd.AddCommand(&cobra.Command{
-		Use:   "list [path]",
-		Short: "List objects in the uwstore",
-		Args:  cobra.RangeArgs(0, 1),
-		RunE:  cmd.StoreListCmd,
+		Use:     "list [path]",
```
Comment (noorvir) [inline main.go:123]: Let's invert this though. Let's add the main command to be `ls` and the alias to be `list`. This should also be done in the root `list` command

Comment (markwinter) [reply main.go]: Done 👍

**Review by noorvir — APPROVED** [review]

**Pull request merged**